### PR TITLE
Adding view and controller for versions. fixes #181

### LINKED ISSUE #181: RollBack Action ends in error
When I upload a file to a work
And I click the RollBack action from the File listing
Then I get an error

screen shot 2015-08-20 at 1 30 54 pm

### PR DESCRIPTION
**Pull request by cam156**
Grabbed the version presenter and action from sufia to implement a version list form.
screen shot 2015-09-01 at 1 08 23 pm

### CODE REVIEW

File: spec/abilities/generic_file_abilities_spec.rb [hunk 1]
```diff
@@ -16,46 +16,43 @@
       describe 'creator of object' do
         let(:creating_user) { user }
         let(:current_user) { user }
-        it do
-          should be_able_to(:create, GenericFile.new)
-          should be_able_to(:read, generic_file)
-          should be_able_to(:update, generic_file)
-          should_not be_able_to(:delete, generic_file)
-        end
+        it { is_expected.to be_able_to(:create, GenericFile.new) }
```
Comment (jcoyne) [inline spec/abilities/generic_file_abilities_spec.rb]: You've transformed 1 spec into 5 here. This is going to cause the test suite to run slower. Are you sure this is necessary?

File: spec/abilities/generic_file_abilities_spec.rb [hunk 2]
```diff
@@ -16,46 +16,43 @@
...
         let(:manager_user) { FactoryGirl.create(:admin) }
         let(:creating_user) { user }
         let(:current_user) { manager_user }
-        it do
-          should be_able_to(:create, ::GenericFile.new)
-          should be_able_to(:read, generic_file)
-          should be_able_to(:update, generic_file)
-          should be_able_to(:destroy, generic_file)
-        end
+        it { is_expected.to be_able_to(:create, GenericFile.new) }
```
Comment (jcoyne) [inline spec/abilities/generic_file_abilities_spec.rb]: You've transformed 1 spec into 5 here. This is going to cause the test suite to run slower. Are you sure this is necessary?

File: spec/abilities/generic_file_abilities_spec.rb [hunk 3]
```diff
@@ -16,46 +16,43 @@
...
         let(:creating_user) { FactoryGirl.create(:user) }
         let(:current_user) { user }
-        it do
-          should be_able_to(:create, ::GenericFile.new)
-          should_not be_able_to(:read, generic_file)
-          should_not be_able_to(:update, generic_file)
-          should_not be_able_to(:delete, generic_file)
-        end
+
+        it { is_expected.to be_able_to(:create, GenericFile.new) }
```
Comment (jcoyne) [inline spec/abilities/generic_file_abilities_spec.rb]: You've transformed 1 spec into 5 here. This is going to cause the test suite to run slower. Are you sure this is necessary?

File: spec/abilities/generic_file_abilities_spec.rb [hunk 4]
```diff
@@ -16,46 +16,43 @@
...
       describe 'a nil user' do
         let(:creating_user) { FactoryGirl.create(:user) }
         let(:current_user) { nil }
-        it do
-          should_not be_able_to(:create, GenericFile.new)
-          should_not be_able_to(:read, generic_file)
-          should_not be_able_to(:update, generic_file)
-          should_not be_able_to(:delete, generic_file)
-        end
+        it { is_expected.to_not be_able_to(:create, GenericFile.new) }
```
Comment (jcoyne) [inline spec/abilities/generic_file_abilities_spec.rb]: You've transformed 1 spec into 5 here. This is going to cause the test suite to run slower. Are you sure this is necessary?

**Comment by cam156** [comment]
@jcoyne I assumed I was going to be asked to transform the tests into rspec3 syntax.  No it is not necessary  it just seems to be the thing we do.

**Comment by cam156** [comment]
@jcoyne pushed the tests back together

File: app/presenters/curation_concerns/version_list_presenter.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    def initialize(version_list)
+      @raw_list = version_list
+    end
+
+    delegate :each, to: :wrapped_list
+
+    private
+
+      def wrapped_list
+        @wrapped_list ||= @raw_list.map { |v| CurationConcerns::VersionPresenter.new(v) }.sort { |a, b| b.version.created <=> a.version.created }.tap { |l| l.first.try(:current!) }
```
Comment (terrellt) [inline app/presenters/curation_concerns/version_list_presenter.rb]: Can we add some indenting or something here at least? It's a lot to take in.

File: app/presenters/curation_concerns/version_list_presenter.rb [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    delegate :each, to: :wrapped_list
+
+    private
+
+      def wrapped_list
+        @wrapped_list ||=
+            @raw_list.map { |v| CurationConcerns::VersionPresenter.new(v) } # wrap each item in a presenter
+            .sort { |a, b| b.version.created <=> a.version.created } # sort list of versions based on creation date
+            .tap { |l| l.first.try(:current!) } # set the first version to the current version
+      end
```
Comment (cam156) [inline app/presenters/curation_concerns/version_list_presenter.rb]: @terrellt Better?

Comment (terrellt) [inline app/presenters/curation_concerns/version_list_presenter.rb]: :+1:

**Comment by terrellt** [comment]
@jcoyne Anything else from your review?

**Pull request merged**